### PR TITLE
Ghosts tracking

### PR DESCRIPTION
**Pull request by simonrozsival**
Ref #1335 

Squash whenever

### CODE REVIEW

**Comment by simonrozsival** [comment]
I updated the base branch to `develop` (this included some cherry-picking and force pushing to this branch).

File: Toggl.Foundation/Sync/States/Pull/CreateGhostProjectsState.cs [hunk 1]
```diff
@@ -34,16 +41,18 @@ public IObservable<ITransition> Start(IFetchObservables fetch)
                 .Distinct(timeEntry => timeEntry.ProjectId)
                 .SelectMany(createGhostForTimeEntryIfProjectIsNotInDatabase)
                 .ToList()
+                .Select(list => list.Count(ghostWasCreated => ghostWasCreated))
+                .Do(analyticsService.TrackCreatingProjectGhosts)
```
Comment (juanlaube) [inline Toggl.Foundation/Sync/States/Pull/CreateGhostProjectsState.cs]: Is the idea to also track an event when no ghost projects are created?
If that's the idea, maybe a test can save some questions in the future 😄

Comment (simonrozsival) [reply Toggl.Foundation/Sync/States/Pull/CreateGhostProjectsState.cs]: This should track `0` when either the fetched list of time entries is empty or when no ghost is created (most of the times) - I can add a test case for that 👍

Comment (juanlaube) [reply Toggl.Foundation/Sync/States/Pull/CreateGhostProjectsState.cs]: If that's the case I don't see the point of adding another tests. :shipit:

File: Toggl.Foundation/Sync/States/Pull/CreateGhostProjectsState.cs [hunk 2]
```diff
@@ -34,16 +41,18 @@ public IObservable<ITransition> Start(IFetchObservables fetch)
                 .Distinct(timeEntry => timeEntry.ProjectId)
                 .SelectMany(createGhostForTimeEntryIfProjectIsNotInDatabase)
                 .ToList()
+                .Select(list => list.Count(ghostWasCreated => ghostWasCreated))
```
Comment (willsb) [inline Toggl.Foundation/Sync/States/Pull/CreateGhostProjectsState.cs]: `Identity`

Comment (simonrozsival) [reply Toggl.Foundation/Sync/States/Pull/CreateGhostProjectsState.cs]: I am not sure in this case - the name of the boolean clearly says what is counted, so you don't have to think about this too much. Although I can rename `list` into something that makes more sense 🤔 give me a sec..

**Pull request merged**